process/smartContract/builtInFunctions/esdtBurn.go

```diff
@@ -62,14 +62,9 @@ func (e *esdtBurn) ProcessBuiltinFunction(
 	e.mutExecution.RLock()
 	defer e.mutExecution.RUnlock()
 
-	if vmInput == nil {
-		return nil, process.ErrNilVmInput
-	}
-	if vmInput.CallValue.Cmp(zero) != 0 {
-		return nil, process.ErrBuiltInFunctionCalledWithValue
-	}
-	if len(vmInput.Arguments) != 2 {
-		return nil, process.ErrInvalidArguments
+	err := checkBasicESDTArguments(vmInput)
```

Comment:
Inside this method you check len(vmInput.Arguments) < 2 and here was checked this way: len(vmInput.Arguments) != 2

Reply:
done